contrib/traceloopctl

```diff
@@ -0,0 +1,131 @@
+#!/bin/sh
+CMD="$1"
+ARG1="$2"
+ARG2="$3"
+set -euo pipefail
+
+if [ $# -lt 1 ] || [ "$CMD" = "-h" ] || [ "$CMD" = "--help" ]; then
+  echo "Usage: $0 COMMAND|-h|--help"
+  echo "Needs to be run with access to /run/traceloop.socket (e.g., with sudo)."
+  echo "Commands:"
+  echo "  list-all"
+  echo "  dump-id ID"
+  echo "  dump-name NAME"
+  echo "  close-id ID"
+  echo "  close-name NAME"
+  echo "  add-current-cgroup NAME"
+  echo
+  echo "  list-sd-units"
+  echo "  list-sd-traces SERVICE"
+  echo "  dump-sd SERVICE INVOCATION|-1"
+  echo "  close-sd SERVICE INVOCATION|-1|all"
+  echo "  add-current-cgroup-sd SERVICE INVOCATION"
+  echo
+  echo "The *-sd commands assume the trace name format is systemd_UNIT_INVOCATIONID which can be automated with:"
+  echo '  ExecStartPre=+/…/contrib/traceloopctl add-current-cgroup-sd "%n" "$INVOCATION_ID"'
+  exit 1
+fi
+
+SCRIPT_FOLDER="$(dirname "$(readlink -f "$0")")"
+CURL="curl -fsS --unix-socket /run/traceloop.socket"
+if [ "$CMD" = list-all ]; then
+  $CURL "http://localhost/list"
+elif [ "$CMD" = dump-id ]; then
+  if [ "$ARG1" = "" ]; then
+    echo "Expected ID argument" > /dev/stderr
+    exit 1
+  fi
+  ID="$ARG1"
+  $CURL "http://localhost/dump?id=${ID}"
+elif [ "$CMD" = dump-name ]; then
+  if [ "$ARG1" = "" ]; then
+    echo "Expected NAME argument" > /dev/stderr
+    exit 1
+  fi
+  NAME="$ARG1"
+  $CURL "http://localhost/dump-by-name?name=${NAME}"
+elif [ "$CMD" = close-id ]; then
+  if [ "$ARG1" = "" ]; then
+    echo "Expected ID argument" > /dev/stderr
+    exit 1
+  fi
+  ID="$ARG1"
+  $CURL "http://localhost/close?id=${ID}"
+elif [ "$CMD" = close-name ]; then
+  if [ "$ARG1" = "" ]; then
+    echo "Expected NAME argument" > /dev/stderr
+    exit 1
+  fi
+  NAME="$ARG1"
+  $CURL "http://localhost/close-by-name?name=${NAME}"
+elif [ "$CMD" = add-current-cgroup ]; then
+    if [ "$ARG1" = "" ]; then
+    echo "Expected NAME argument" > /dev/stderr
+    exit 1
+  fi
+  NAME="$ARG1"
+  CURRENT_CGROUP=$("${SCRIPT_FOLDER}"/current-cgroup)
+  $CURL "http://localhost/add?name=${NAME}&cgrouppath=${CURRENT_CGROUP}"
+elif [ "$CMD" = list-sd-units ]; then
+  echo " Traces Units"
+  echo "------- -----"
+  $CURL "http://localhost/list" | grep -o '\[systemd_.*\]' | cut -d _ -f 2 | sort | uniq -c
```

Comment:
The systemd-specific commands *-sd could be added in the Go program, so that the shell script could be simplified. (could be in a follow-up PR)

Reply:
And internally traceloop with still concatenate a special name or would you introduce a new attributes for Tracelet?

Reply:
I guess introducing new attributes but I have not given it much thought. Do you have a preference?